### PR TITLE
WIP Add better support for multiple flavors

### PR DESCRIPTION
**Pull request by marek-siemieniuk-morawski**
Changes I made:

- Depending on the flavor, env variables are being read from `.env.beta`, `.env.prod`
- Depending on the flavor, `google-services.json` are being loaded from `android/app/src/beta` or `android/app/src/beta` (not done for iOS yet)
- Added files that include sensitive (API keys, tokens) data to `.gitignore`
- Since now the Beta version names `GoldWallet Beta` instead of `Testnet GoldWallet`
- Added a valid Sentra DNS to `sentry.properties` files and since now it mustn't be stored as an environment variable
- Fixed running & building Release versions on local machines
- Since now the release versions are signed during the build process (`/gradlew assembleProd`)
- Since now it's possible to build both Prod and Beta versions of the iOS app
- Since now Detox tests work with both iOS and Android

What left:
- Update README.md 
- Update Bitrise pipelines 
- Add support for multiple `GoogleService-Info.plist` files
- Add a script that fetches required files from private repo*

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -17,7 +17,6 @@ const i18n = require('./loc');
 
 if (!__DEV__) {
   Sentry.init({
-    dsn: config.sentryDsn,
```
Comment (marek-siemieniuk-morawski) [inline App.tsx:20]: Removing that makes that DNS is fetched from `sentry.properties` file

File: config/index.js [hunk 2]
```diff
@@ -2,21 +2,16 @@ import Config from 'react-native-config';
 
 const bitcoin = require('bitcoinjs-lib');
 
-const { SENTRY_DSN, HOST, PORT, BTCV_NETWORK, PROTOCOL, ELECTRUM_X_PROTOCOL_VERSION, IS_BETA } = Config;
-
-let isBeta = false;
-try {
-  isBeta = JSON.parse(IS_BETA);
-} catch (_) {}
+const { SENTRY_DSN, HOST, PORT, BTCV_NETWORK, PROTOCOL, ELECTRUM_X_PROTOCOL_VERSION, IS_BETA, APP_ID, APPLICATION_NAME } = Config;
```
Comment (marek-siemieniuk-morawski) [inline config/index.js:6]: IMO this file should be removed at all in the next PR

**Review by Kamill90 — APPROVED** [review]
works locally for me

File: config/index.js (hunk 2 above)
Comment (KedziaPawel) [reply config/index.js:6]: what do you want to use instead?

Comment (marek-siemieniuk-morawski) [reply config/index.js:6]: Wherever it's required, simply import `'react-native-config'` and use the env variable directly

Comment (KedziaPawel) [reply config/index.js:6]: I prefer to  have this file as it gives a level of abstraction, we can parse env vars here, add default one, and config that doesn't depend on env var

Comment (marek-siemieniuk-morawski) [reply config/index.js:6]: Okay then, if you find it useful let's keep it. It actually makes sense 👍

**Review by KedziaPawel — APPROVED** [review]
works for me on android and ios

**Review by aleparty — APPROVED** [review]
works for me as well

**Review by aleparty — APPROVED** [review]

**Review by mateuszpmroz — APPROVED** [review]

**Comment by mateuszpmroz** [comment]
Working fine on android

**Pull request closed without merging**